codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -629,7 +631,12 @@ public void operationComplete(ChannelFuture future) throws Exception {
 
     @Override
     public ChannelFuture resetStream(final ChannelHandlerContext ctx, int streamId, long errorCode,
-                                     final ChannelPromise promise) {
+                                     ChannelPromise promise) {
+        if (promise.isVoid()) {
+            // An error writing RST_STREAM closes the connection eventually, however we need to know before
+            // what error occurred.
+            promise = promise.unvoid().addListener(Http2CodecUtil.PIPELINE_EXCEPTION_ON_ERROR_LISTENER);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'Http2CodecUtil.PIPELINE_EXCEPTION_ON_ERROR_LISTENER' due to existing static import 'io.netty.handler.codec.http2.Http2CodecUtil.PIPELINE_EXCEPTION_ON_ERROR_LISTENER' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)